conda_smithy/configure_feedstock.py

```diff
@@ -319,10 +374,27 @@ def _trim_unused_pin_run_as_build(all_used_vars):
 
 
 def _get_used_key_values_by_input_order(
-    squished_input_variants,
-    squished_used_variants,
-    all_used_vars,
-):
+    squished_input_variants: Union[
+        OrderedDict,
+        Dict[
+            str,
+            Union[
+                Set[str],
+                Dict[str, Dict[str, str]],
+                List[str],
+                Tuple[str, str, str, str],
+                Tuple[str],
+                Tuple[str, str, str, str, str],
+                List[List[str]],
```

Comment:
These tuples look odd -- this really looks like the function really expects an arbitrary-size collection of strings (`List[str]`, `Set[str]` etc) but some callers are passing this as a tuple. Rather than specifying every size of tuple we accept I would treat this as a signal to go and fix the callers to stop them using tuples.

Reply:
i changed it to this `Union[set, dict, list, tuple]` keeping it generic.. not sure if this helps to reduce complexity..